src/graph/topo.cc

```diff
@@ -607,62 +607,92 @@ ncclResult_t ncclTopoGetLocalNet(struct ncclTopoSystem* system, int rank, int64_
   return ncclSuccess;
 }
 
-ncclResult_t scklGetTopoFromXMLAndSetChannels(struct ncclComm* comm) {
+ncclResult_t scklGetAlgoFromXMLAndSetComm(struct ncclComm* comm) {
   char* str = getenv("SCKL_XML_FILE");
   if (str){
     INFO(NCCL_ENV, "SCKL_XML_FILE set by environment to %s", str);
     struct ncclXml* xml;
     NCCLCHECK(ncclCalloc(&xml, 1));
-    NCCLCHECK(scklTopoGetXmlGraphFromFile(str, xml));
+    NCCLCHECK(scklGetXmlAlgoFromFile(str, xml));
     int rank = comm->rank;
 
-    for (int c=0; c<comm->nChannels; c++){
-      comm->channels[c].sGraph.nRecvPeers = 0;
-      comm->channels[c].sGraph.nSendPeers = 0;
-    }
-    
+    struct scklAlgorithm* scklAlgo = &comm->scklAlgo;
+    // zeroing out all entries.
+    memset(scklAlgo, 0, sizeof(struct scklAlgorithm));
     struct ncclXmlNode* topNode;
-    NCCLCHECK(xmlFindTag(xml, "system", &topNode));
+    NCCLCHECK(xmlFindTag(xml, "algo", &topNode));
+    int nchunks;
+    NCCLCHECK(xmlGetAttrInt(topNode, "nchunks", &nchunks));
```

Comment:
To be consistent with sckl-core `nchunks` should be `naddresses` or something to that effect.

Reply:
For alltoall, the total number of chunks in the system is nchunks * nranks^2.